lib/terraform.sh

```diff
@@ -12,10 +12,11 @@ function run_terraform_command() {
     # Initialize terraform w/remote AWS backend configuration
 
     terraform init \
-        -backend-config="bucket=$TERRAFORM_S3_BUCKET" \
-        -backend-config="key=$TERRAFORM_S3_KEY" \
+        -backend="$TERRAFORM_BACKEND"
```

Comment:
For one, I think this PR is too large and would prefer if backend setup moved to another PR.

Also, don't think this is the best way... I think we should default to "local" instead of skipping backend config altogether.

See: https://www.terraform.io/docs/commands/init.html#backend-initialization

> To skip backend configuration, use -backend=false. Note that some other init steps require an initialized backend, so it is recommended to use this flag only when the working directory was already previously initialized for a particular backend.